src/rapids_dependency_file_generator/rapids_dependency_file_validator.py

```diff
@@ -1,137 +1,13 @@
-                                        "then": {"items": {"type": "string"}},
-                                    },
-                                    "matrices": {
-                                        "type": "array",
-                                        "items": {
-                                            "type": "object",
-                                            "properties": {
-                                                "matrix": {"type": ["null", "object"]},
-                                                "packages": {
-                                                    "type": ["null", "array", "string"],
-                                                    "if": {"type": "array"},
-                                                    "then": {
-                                                        "items": {
-                                                            "type": ["string", "object"]
-                                                        },
-                                                        "if": {"type": "object"},
-                                                        "then": {
-                                                            "patternProperties": {
-                                                                ".*": {
-                                                                    "type": [
-                                                                        "array",
-                                                                        "string",
-                                                                    ],
-                                                                    "if": {
-                                                                        "type": "array"
-                                                                    },
-                                                                    "then": {
-                                                                        "items": {
-                                                                            "type": "string"
-                                                                        }
-                                                                    },
-                                                                }
-                                                            }
-                                                        },
-                                                    },
-                                                },
-                                            },
-                                        },
-                                    },
-                                },
-                            },
-                        },
-                    },
-                }
-            },
-        },
-    },
-    "required": ["files", "dependencies"],
-}
+SCHEMA = json.loads(pkg_resources.resource_string(__name__, "schema.json"))
```

Comment:
Is there a reason to prefer this to a simple `with open...: json.load(...)`? Not critical, I'm just not familiar with this API and it seems like it's effectively using the same path information via `__name__` instead of `__file__` (which I suppose one could argue is cleaner).

Reply:
I assume that this will work more universally, e.g., when the package is loaded directly from a zip-file or so, but that's something I could look into if you like. By using this function we don't have to worry about the internal structure of the package I hope.

Reply:
I'm fine leaving it as is. I don't know what cases it will handle better, but I'm sure you're right that there are some.